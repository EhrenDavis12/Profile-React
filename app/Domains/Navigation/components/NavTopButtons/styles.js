import styled from 'styled-components';
import ThemeFactory from 'Domains/SharedStyles/ThemeFactory';
import BasicButton from 'Domains/SharedKernel/components/Buttons/BasicButton';

import { Link } from 'react-scroll';
const theme = ThemeFactory();

export const ItemButtonStyled = styled(BasicButton)`
  padding: 25px;
  text-transform: uppercase;
  cursor: pointer;
  border-bottom: solid 1px ${theme.SecondaryBkg};

  &:hover {
    background-color: ${theme.SecondaryBkg};
  }
`;

export const LinkStyled = styled(Link)`
  padding: 25px;
  text-transform: uppercase;
  cursor: pointer;
  border-bottom: solid 1px ${theme.SecondaryBkg};

  &:hover {
    background-color: ${theme.SecondaryBkg};
  }
`;

// display: ${props => (props.isDrawerOpen ? 'none' : 'true')};
export const HamburgerButtonStyled = styled(BasicButton)``;

export const Nav = styled.nav`
  padding-left: 0px;
  z-index: 900;
`;
