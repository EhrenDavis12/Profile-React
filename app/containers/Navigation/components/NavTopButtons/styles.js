import styled from 'styled-components';
import ThemeFactory from 'components/SharedStyles/ThemeFactory';
import BasicButton from 'components/Buttons/BasicButton';

import BannerImg from 'images/sandpaper.png';

import { Link } from 'react-scroll';
const theme = ThemeFactory();

// export const ItemButtonStyled = styled(BasicButton)`
//   padding: 25px;
//   text-transform: uppercase;

//   &:hover {
//     background-color: ${theme.SecondaryBkg};
//   }
// `;

export const LinkStyled = styled(Link)`
  padding: 15px 25px 15px 25px;
  text-transform: uppercase;
  cursor: pointer;
  // background-color: red;
  hight: 100px;

  &:hover {
    background-color: ${theme.SecondaryBkg};
  }
`;

// display: ${props => (props.isDrawerOpen ? 'none' : 'true')};
export const HamburgerButtonStyled = styled(BasicButton)``;

export const Nav = styled.nav`
  padding: 15px 0px 15px 20px;
  z-index: 900;
  background-image: url('${BannerImg}');
  background-color: ${theme.SecondaryBkg};
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
`;
