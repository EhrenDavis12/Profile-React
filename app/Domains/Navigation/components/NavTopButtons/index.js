/**
 *
 * AppBar
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
// import { animateScroll as scroll } from 'react-scroll';
// import FontAwesome from 'react-fontawesome';
// import BasicButton from 'Domains/SharedKernel/components/Buttons/BasicButton';
import { Nav, HamburgerButtonStyled, LinkStyled } from './styles';
// import { Nav } from './styles';

function NavTopButtons({ toggleDrawer, items, itemLabelAttr, itemKeyAttr }) {
  /* scrollToTop = () => {
    scroll.scrollToTop();
  }; */

  const itemLinkAttr = 'LinkTo';
  const True = true; // eslint errors smooth=true
  const itemNodes = items.map(item => (
    <li className="nav-item" key={`${item[itemKeyAttr]}"_nav"`}>
      <LinkStyled
        activeClass="active"
        smooth={True}
        duration={600}
        to={`${item[itemLinkAttr]}`}
        offset={-70}
      >
        {item[itemLabelAttr]}
      </LinkStyled>
    </li>
  ));

  return (
    <Nav className="navbar sticky-top navbar-expand-md navbar-light">
      <HamburgerButtonStyled
        onClick={toggleDrawer}
        className="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbarNav"
        aria-controls="navbarNav"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span className="navbar-toggler-icon" />
      </HamburgerButtonStyled>
      <div className="collapse navbar-collapse" id="navbarNav">
        <ul className="nav justify-content-end">{itemNodes}</ul>
      </div>
    </Nav>
  );
}

NavTopButtons.propTypes = {
  toggleDrawer: PropTypes.func.isRequired,
  items: PropTypes.array.isRequired,
  itemLabelAttr: PropTypes.string.isRequired,
  itemKeyAttr: PropTypes.string.isRequired,
};

export default NavTopButtons;
