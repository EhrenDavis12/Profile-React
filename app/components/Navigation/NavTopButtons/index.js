/**
 *
 * AppBar
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
// import { animateScroll as scroll } from 'react-scroll';
// import FontAwesome from 'react-fontawesome';
import { Nav, HamburgerButtonStyled, LinkStyled } from './styles';

function NavTopButtons({ toggleDrawer, items, itemLabelAttr, itemKeyAttr }) {
  /* scrollToTop = () => {
    scroll.scrollToTop();
  }; */

  const itemLinkAttr = 'LinkTo';
  const True = true; // eslint errors smooth=true
  const itemNodes = items.map(item => (
    <li className="nav-item" key={`${item[itemKeyAttr]}"_nav"`}>
      <LinkStyled
        activeClass="active"
        smooth={True}
        duration={600}
        to={`${item[itemLinkAttr]}`}
        offset={-70}
      >
        {item[itemLabelAttr]}
      </LinkStyled>
    </li>
  ));

  return (
    // <Nav className="navbar navbar-expand-md navbar-dark fixed-top bg-dark">
    <Nav className="navbar fixed-top navbar-expand-lg navbar-light">
      <HamburgerButtonStyled
        onClick={toggleDrawer}
        className="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbarNav"
        aria-controls="navbarNav"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span className="navbar-toggler-icon" />
      </HamburgerButtonStyled>
      <div className="collapse navbar-collapse" id="navbarNav">
        <ul className="nav">{itemNodes}</ul>
      </div>
    </Nav>
  );
}

NavTopButtons.propTypes = {
  toggleDrawer: PropTypes.func.isRequired,
  items: PropTypes.array.isRequired,
  itemLabelAttr: PropTypes.string.isRequired,
  itemKeyAttr: PropTypes.string.isRequired,
};

export default NavTopButtons;
