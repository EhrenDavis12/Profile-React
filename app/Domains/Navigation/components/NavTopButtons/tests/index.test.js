import React from 'react';
import 'jest-styled-components';
import { mount } from 'enzyme';
// import { LinkStyled } from '../styles';

import NavTopButtons from '../index';

const testData = {
  toggleDrawer: () => true,
  items: [
    {
      key: 'AboutMe',
      LinkTo: 'AboutMe',
      DisplayName: 'About Me',
    },
    {
      key: 'Portfolio',
      LinkTo: 'portfolio',
      DisplayName: 'Portfolio',
    },
  ],
  selectItem: () => true,
  itemLabelAttr: 'DisplayName',
  itemKeyAttr: 'key',
  isDrawerOpen: false,
};

const renderComponent = mockData => mount(<NavTopButtons {...mockData} />);

describe('<NavTopButtons />', () => {
  it('renders as expected', () => {
    const wrapper = renderComponent(testData);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have children', () => {
    const wrapper = renderComponent(testData);
    expect(wrapper.find('ul').prop('children').length).toEqual(2);
  });

  /* it('Children should handle click events', () => {
    const mockCallBack = jest.fn(() => true);
    const propsData = {
      ...testData,
      selectItem: mockCallBack,
      isDrawerOpen: true,
    };
    const renderedComponent = renderComponent(propsData);
    renderedComponent
      .find(LinkStyled)
      .first()
      .simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  }); */
});
