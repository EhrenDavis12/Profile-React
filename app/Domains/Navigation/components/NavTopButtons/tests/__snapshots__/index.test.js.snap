// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavTopButtons /> renders as expected 1`] = `
.c1 {
  padding: 25px;
  text-transform: uppercase;
  cursor: pointer;
  border-bottom: solid 1px #f6f6f6;
}

.c1:hover {
  background-color: #f6f6f6;
}

.c0 {
  padding-left: 0px;
  z-index: 900;
}

<NavTopButtons
  isDrawerOpen={false}
  itemKeyAttr="key"
  itemLabelAttr="DisplayName"
  items={
    Array [
      Object {
        "DisplayName": "About Me",
        "LinkTo": "AboutMe",
        "key": "AboutMe",
      },
      Object {
        "DisplayName": "Portfolio",
        "LinkTo": "portfolio",
        "key": "Portfolio",
      },
    ]
  }
  selectItem={[Function]}
  toggleDrawer={[Function]}
>
  <styles__Nav
    className="navbar sticky-top navbar-expand-md navbar-light"
  >
    <StyledComponent
      className="navbar sticky-top navbar-expand-md navbar-light"
      forwardedClass={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "styles__Nav-sc-1iujtvx-3",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "padding-left:0px;z-index:900;",
            ],
          },
          "displayName": "styles__Nav",
          "render": [Function],
          "styledComponentId": "styles__Nav-sc-1iujtvx-3",
          "target": "nav",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <nav
        className="navbar sticky-top navbar-expand-md navbar-light c0"
      >
        <styles__HamburgerButtonStyled
          aria-controls="navbarNav"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-target="#navbarNav"
          data-toggle="collapse"
          onClick={[Function]}
          type="button"
        >
          <StyledComponent
            aria-controls="navbarNav"
            aria-expanded="false"
            aria-label="Toggle navigation"
            className="navbar-toggler"
            data-target="#navbarNav"
            data-toggle="collapse"
            forwardedClass={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": undefined,
                "componentStyle": ComponentStyle {
                  "componentId": "styles__HamburgerButtonStyled-sc-1iujtvx-2",
                  "isStatic": true,
                  "lastClassName": "dqgFGn",
                  "rules": Array [],
                },
                "displayName": "styles__HamburgerButtonStyled",
                "render": [Function],
                "styledComponentId": "styles__HamburgerButtonStyled-sc-1iujtvx-2",
                "target": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              }
            }
            forwardedRef={null}
            onClick={[Function]}
            type="button"
          >
            <BasicButton
              aria-controls="navbarNav"
              aria-expanded="false"
              aria-label="Toggle navigation"
              className="navbar-toggler "
              data-target="#navbarNav"
              data-toggle="collapse"
              onClick={[Function]}
              type="button"
            >
              <div
                className="navbar-toggler "
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  className="navbar-toggler-icon"
                  key=".0"
                />
              </div>
            </BasicButton>
          </StyledComponent>
        </styles__HamburgerButtonStyled>
        <div
          className="collapse navbar-collapse"
          id="navbarNav"
        >
          <ul
            className="nav justify-content-end"
          >
            <li
              className="nav-item"
              key="AboutMe\\"_nav\\""
            >
              <styles__LinkStyled
                activeClass="active"
                duration={600}
                offset={-70}
                smooth={true}
                to="AboutMe"
              >
                <StyledComponent
                  activeClass="active"
                  duration={600}
                  forwardedClass={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": undefined,
                      "componentStyle": ComponentStyle {
                        "componentId": "styles__LinkStyled-sc-1iujtvx-1",
                        "isStatic": true,
                        "lastClassName": "c1",
                        "rules": Array [
                          "padding:25px;text-transform:uppercase;cursor:pointer;border-bottom:solid 1px ",
                          "#f6f6f6",
                          ";&:hover{background-color:",
                          "#f6f6f6",
                          ";}",
                        ],
                      },
                      "displayName": "styles__LinkStyled",
                      "render": [Function],
                      "styledComponentId": "styles__LinkStyled-sc-1iujtvx-1",
                      "target": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                      Symbol(Symbol.hasInstance): [Function],
                    }
                  }
                  forwardedRef={null}
                  offset={-70}
                  smooth={true}
                  to="AboutMe"
                >
                  <Link
                    activeClass="active"
                    className="c1"
                    duration={600}
                    offset={-70}
                    smooth={true}
                    to="AboutMe"
                  >
                    <LinkElement
                      className="c1"
                      onClick={[Function]}
                    >
                      <a
                        className="c1"
                        onClick={[Function]}
                      >
                        About Me
                      </a>
                    </LinkElement>
                  </Link>
                </StyledComponent>
              </styles__LinkStyled>
            </li>
            <li
              className="nav-item"
              key="Portfolio\\"_nav\\""
            >
              <styles__LinkStyled
                activeClass="active"
                duration={600}
                offset={-70}
                smooth={true}
                to="portfolio"
              >
                <StyledComponent
                  activeClass="active"
                  duration={600}
                  forwardedClass={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": undefined,
                      "componentStyle": ComponentStyle {
                        "componentId": "styles__LinkStyled-sc-1iujtvx-1",
                        "isStatic": true,
                        "lastClassName": "c1",
                        "rules": Array [
                          "padding:25px;text-transform:uppercase;cursor:pointer;border-bottom:solid 1px ",
                          "#f6f6f6",
                          ";&:hover{background-color:",
                          "#f6f6f6",
                          ";}",
                        ],
                      },
                      "displayName": "styles__LinkStyled",
                      "render": [Function],
                      "styledComponentId": "styles__LinkStyled-sc-1iujtvx-1",
                      "target": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                      Symbol(Symbol.hasInstance): [Function],
                    }
                  }
                  forwardedRef={null}
                  offset={-70}
                  smooth={true}
                  to="portfolio"
                >
                  <Link
                    activeClass="active"
                    className="c1"
                    duration={600}
                    offset={-70}
                    smooth={true}
                    to="portfolio"
                  >
                    <LinkElement
                      className="c1"
                      onClick={[Function]}
                    >
                      <a
                        className="c1"
                        onClick={[Function]}
                      >
                        Portfolio
                      </a>
                    </LinkElement>
                  </Link>
                </StyledComponent>
              </styles__LinkStyled>
            </li>
          </ul>
        </div>
      </nav>
    </StyledComponent>
  </styles__Nav>
</NavTopButtons>
`;
